src/shogun/mathematics/linalg/LinalgBackendBase.h

```diff
@@ -0,0 +1,61 @@
+#include <shogun/lib/config.h>
+#include <shogun/lib/common.h>
+#include <shogun/lib/SGVector.h>
+#include <shogun/io/SGIO.h>
+#include <shogun/mathematics/linalg/GPUMemoryBase.h>
+
+#include <iostream>
+
+#ifndef Linalg_Backend_Base_H__
+#define Linalg_Backend_Base_H__
+
+#ifdef HAVE_CXX11
+
+namespace shogun
+{
+
+class LinalgBackendBase
+{
+public:
+    // macro to avoid templating
+	#define BACKEND_GENERIC_DOT(Type) \
```

Comment:
nooooooooooooooooooooooooooup. c++11 has waaay better templating engine to avoid this shitty macro ;) ping me on irc